CIP-0118/README-implicit.md

```diff
@@ -0,0 +1,895 @@
+to determine whether any intent _will_ ever be fulfilled, or even whether it was
+created with a malicious intent (spam).
+
+**On-Chain (Mempool)**.
+The possibility of un-fulfillable intents is an important reason for
+transmitting only entire zones across the Cardano network. Upon validating an
+incoming zone, a node is able to
+determine whether to discard it or add it to the mempool. This is not the
+case for intent-containing transactions. If we were to allow the mempool to
+contain intent-containing transactions outside zones, such a transaction
+could be placed in a mempool indefinitely, waiting for an appropriate
+partially-valid counterpart to complete the zone. An attacker could fill up
+a node's mempool with such transactions, rendering it unable to record
+additional incoming transactions. Zone-based validation
+prevents this attack.
+
+**On-Chain (Phase-1)**.
+Since zones are size-restricted by the constraint that currently restricts the
+size of a single transaction (`maxTxSize`), we assume that the phase-1 validation cost
+has the same limitations as before. So, phase-1 validation of a single
+zone is likely to cost the same as phase-1 validation of a single transaction
+currently costs. Therefore, we conjecture that the risks of a spam attack using phase-1
+invalid entities is not increased as compared to what it currently is.
+
+**On-Chain (Phase-2)**. The mempool attack scenario describes mitigation of an attack
+which results in a DDoS by using up all the node's mempool storage. Another kind of
+DDoS attack which we must defend against is one that involves a malicious
+party sending transactions with large (i.e. memory- or CPU-use-intensive)
+failing scripts. We refer to a script failure in a transaction as a _phase-2_ failure.
+
+In the current ledger design, this is mitigated by requiring a transaction to provide
+an amount of Ada collateral that is proportional to the sum of the sizes of all
+its scripts. This collateral is collected in the case of script failure.
+A phase-2 transaction failure in the design proposed in this CIP may result in phase-1 validation
+failures of subsequent transactions.
+To address this problem, we have defined the `ZONE-N` rule of `ZONE`. It requires
+that the collateral of each transaction
+
+- comes from the UTxO set to which the _zone_, rather than the transaction, is applied
+- covers the cost of running all scripts in all preceding transactions (itself included)
+
+This guarantees that enough collateral can
+always be collected the last transaction in a zone to cover all scripts that have been
+run during block validation as well as during mempool validation, compensating nodes
+for their validation work.
+
+The `ZONE-N` rule only applies whenever there is exactly one invalid transaction
+in the list, and it is the last one. It is not the case that we expect that a zone will always phase-2 fail at
+the last transaction. Rather, a mempool should stop validating and drop transactions
```

Comment:
mempool is not capable at this point of doing anything like that, so this type of communication between ledger and consensus will need to be invented